### PR TITLE
Import XPath performance

### PR DESCRIPTION
**Pull request by anthonyccri**
Use CachedXPathAPI to improve xpath performance in xml converter

### CODE REVIEW

File: geomesa-convert/geomesa-convert-xml/src/main/scala/org/locationtech/geomesa/convert/xml/XMLConverter.scala [hunk 1]
```diff
@@ -58,7 +59,10 @@ class XMLConverter(val targetSFT: SimpleFeatureType,
     val root = docBuilder.parse(new InputSource(new StringReader(i))).getDocumentElement
     featurePath.map { path =>
       val nodeList = path.evaluate(root, XPathConstants.NODESET).asInstanceOf[NodeList]
-      (0 until nodeList.getLength).map(i => Array[Any](nodeList.item(i)))
+        (0 until nodeList.getLength).map { i =>
+          val item = nodeList.item(i)
+          Array[Any](item)
+        }.toSeq
```
Comment (jnh5y) [inline geomesa-convert/geomesa-convert-xml/src/main/scala/org/locationtech/geomesa/convert/xml/XMLConverter.scala]: (style) I prefer the original.  If a seq is needed, I'd suggest
```
(0 until nodeList.getLength).map(i => Array[Any](nodeList.item(i))).toSeq
```

File: geomesa-convert/geomesa-convert-xml/pom.xml [hunk 2]
```diff
@@ -19,6 +19,12 @@
     <name>GeoMesa Convert XML</name>
 
     <dependencies>
+<dependency>
+    <groupId>xalan</groupId>
+    <artifactId>xalan</artifactId>
+    <version>2.7.2</version>
```
Comment (jnh5y) [inline geomesa-convert/geomesa-convert-xml/pom.xml]: Is this dependency required?  If so, we can piggyback on https://dev.eclipse.org/ipzilla/show_bug.cgi?id=9587.

File: geomesa-convert/geomesa-convert-xml/src/main/scala/org/locationtech/geomesa/convert/xml/XMLConverter.scala [hunk 3]
```diff
@@ -50,15 +50,17 @@ class XMLConverter(val targetSFT: SimpleFeatureType,
...
     // if a schema is defined, validate it - this will throw an exception on failure
     validator.foreach(_.validate(new StreamSource(new StringReader(i))))
     // parse the document once, then extract each feature node and operate on it
     val root = docBuilder.parse(new InputSource(new StringReader(i))).getDocumentElement
+
     featurePath.map { path =>
-      val nodeList = path.evaluate(root, XPathConstants.NODESET).asInstanceOf[NodeList]
-      (0 until nodeList.getLength).map(i => Array[Any](nodeList.item(i)))
+      val nl = cachedAPI.eval(root, path).nodelist()
+      (0 until nl.getLength).map { i => Array[Any](nl.item(i)) }
```
Comment (jnh5y) [inline geomesa-convert/geomesa-convert-xml/src/main/scala/org/locationtech/geomesa/convert/xml/XMLConverter.scala]: Is nodelist array backed?  i.e., is getLength 'cheap'.

**Comment by elahrvivaz** [comment]
looks good

**Comment by jahhulbert-ccri** [comment]
+1

**Comment by jnh5y** [comment]
@jahhulbert-ccri is this branch the 'right' way?  I.e. using xalan?  If so, can you enter a piggyback CQ?

**Pull request closed without merging**